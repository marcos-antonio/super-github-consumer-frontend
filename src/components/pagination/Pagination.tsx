import React, { FunctionComponent } from 'react';

import * as S from './styled';

export interface PaginationProps {
  previousPageDisabled?: boolean;
  nextPageDisabled?: boolean;
  previousPageClickHandler?: () => void;
  nextPageClickHandler?: () => void;
}

export const Pagination: FunctionComponent<PaginationProps> = ({
  previousPageDisabled = false,
  nextPageDisabled = false,
  previousPageClickHandler,
  nextPageClickHandler,
}) => {
  return (
    <S.Container>
      <S.ArrowLeftIcon
        data-testid={'arrow-left-icon'}
        disabled={previousPageDisabled}
        onClick={() => !previousPageDisabled && previousPageClickHandler()}
      />
      <S.ArrowRightIcon
        data-testid={'arrow-right-icon'}
        disabled={nextPageDisabled}
        onClick={() => !nextPageDisabled && nextPageClickHandler()}
      />
    </S.Container>
  );
};
