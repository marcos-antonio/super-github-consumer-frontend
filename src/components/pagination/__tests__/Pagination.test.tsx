import React from 'react';
import { render, RenderResult, fireEvent, wait } from '@testing-library/react';

import { ThemeProvider } from 'styled-components';

import theme from 'components/Theme';

import { Pagination } from '../Pagination';

describe('Component: Pagination', () => {
  let wrapper: RenderResult;
  const nextPageClickMockFunction = jest.fn();
  const previousPageClickMockFunction = jest.fn();

  beforeEach(() => {
    wrapper = render(
      <ThemeProvider theme={theme}>
        <Pagination
          nextPageClickHandler={nextPageClickMockFunction}
          previousPageClickHandler={previousPageClickMockFunction}
        />
      </ThemeProvider>
    );
  });

  it('render the component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should click on next page', async () => {
    const { getByTestId } = wrapper;
    const nextPageIcon = getByTestId('arrow-right-icon');

    await wait(() => fireEvent.click(nextPageIcon));

    expect(nextPageClickMockFunction).toHaveBeenCalled();
  });

  it('should click on previous page', async () => {
    const { getByTestId } = wrapper;
    const nextPageIcon = getByTestId('arrow-left-icon');

    await wait(() => fireEvent.click(nextPageIcon));

    expect(previousPageClickMockFunction).toHaveBeenCalled();
  });
});

describe('Component: Pagination with disabled icons', () => {
  let wrapper: RenderResult;
  const nextPageClickMockFunction = jest.fn();
  const previousPageClickMockFunction = jest.fn();

  beforeEach(() => {
    wrapper = render(
      <ThemeProvider theme={theme}>
        <Pagination
          nextPageDisabled={true}
          previousPageDisabled={true}
          nextPageClickHandler={nextPageClickMockFunction}
          previousPageClickHandler={previousPageClickMockFunction}
        />
      </ThemeProvider>
    );
  });

  it('render the component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should click on next page', async () => {
    const { getByTestId } = wrapper;
    const nextPageIcon = getByTestId('arrow-right-icon');

    await wait(() => fireEvent.click(nextPageIcon));

    expect(nextPageClickMockFunction).not.toHaveBeenCalled();
  });

  it('should click on previous page', async () => {
    const { getByTestId } = wrapper;
    const nextPageIcon = getByTestId('arrow-left-icon');

    await wait(() => fireEvent.click(nextPageIcon));

    expect(previousPageClickMockFunction).not.toHaveBeenCalled();
  });
});
