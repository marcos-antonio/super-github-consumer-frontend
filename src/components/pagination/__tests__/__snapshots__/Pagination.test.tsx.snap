// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Pagination render the component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="styled__Container-q50she-0 flkxSk"
      >
        <svg
          aria-hidden="true"
          class="sc-AxjAm gXIMwt styled__ArrowLeftIcon-q50she-1 hCEWFs"
          data-testid="arrow-left-icon"
          fill="currentColor"
          focusable="false"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M15.41 16.59L10.83 12l4.58-4.59L14 6l-6 6 6 6 1.41-1.41z"
          />
          <path
            d="M0 0h24v24H0V0z"
            fill="none"
          />
        </svg>
        <svg
          aria-hidden="true"
          class="sc-AxjAm gXIMwt styled__ArrowRightIcon-q50she-2 fOntrf"
          data-testid="arrow-right-icon"
          fill="currentColor"
          focusable="false"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8.59 16.59L13.17 12 8.59 7.41 10 6l6 6-6 6-1.41-1.41z"
          />
          <path
            d="M0 0h24v24H0V0z"
            fill="none"
          />
        </svg>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="styled__Container-q50she-0 flkxSk"
    >
      <svg
        aria-hidden="true"
        class="sc-AxjAm gXIMwt styled__ArrowLeftIcon-q50she-1 hCEWFs"
        data-testid="arrow-left-icon"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M15.41 16.59L10.83 12l4.58-4.59L14 6l-6 6 6 6 1.41-1.41z"
        />
        <path
          d="M0 0h24v24H0V0z"
          fill="none"
        />
      </svg>
      <svg
        aria-hidden="true"
        class="sc-AxjAm gXIMwt styled__ArrowRightIcon-q50she-2 fOntrf"
        data-testid="arrow-right-icon"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8.59 16.59L13.17 12 8.59 7.41 10 6l6 6-6 6-1.41-1.41z"
        />
        <path
          d="M0 0h24v24H0V0z"
          fill="none"
        />
      </svg>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Component: Pagination with disabled icons render the component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="styled__Container-q50she-0 flkxSk"
      >
        <svg
          aria-hidden="true"
          class="sc-AxjAm gXIMwt styled__ArrowLeftIcon-q50she-1 eRtPcw"
          data-testid="arrow-left-icon"
          disabled=""
          fill="currentColor"
          focusable="false"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M15.41 16.59L10.83 12l4.58-4.59L14 6l-6 6 6 6 1.41-1.41z"
          />
          <path
            d="M0 0h24v24H0V0z"
            fill="none"
          />
        </svg>
        <svg
          aria-hidden="true"
          class="sc-AxjAm gXIMwt styled__ArrowRightIcon-q50she-2 gMOTVD"
          data-testid="arrow-right-icon"
          disabled=""
          fill="currentColor"
          focusable="false"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8.59 16.59L13.17 12 8.59 7.41 10 6l6 6-6 6-1.41-1.41z"
          />
          <path
            d="M0 0h24v24H0V0z"
            fill="none"
          />
        </svg>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="styled__Container-q50she-0 flkxSk"
    >
      <svg
        aria-hidden="true"
        class="sc-AxjAm gXIMwt styled__ArrowLeftIcon-q50she-1 eRtPcw"
        data-testid="arrow-left-icon"
        disabled=""
        fill="currentColor"
        focusable="false"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M15.41 16.59L10.83 12l4.58-4.59L14 6l-6 6 6 6 1.41-1.41z"
        />
        <path
          d="M0 0h24v24H0V0z"
          fill="none"
        />
      </svg>
      <svg
        aria-hidden="true"
        class="sc-AxjAm gXIMwt styled__ArrowRightIcon-q50she-2 gMOTVD"
        data-testid="arrow-right-icon"
        disabled=""
        fill="currentColor"
        focusable="false"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8.59 16.59L13.17 12 8.59 7.41 10 6l6 6-6 6-1.41-1.41z"
        />
        <path
          d="M0 0h24v24H0V0z"
          fill="none"
        />
      </svg>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
